src/components/Prescriptions.tsx

```diff
@@ -42,35 +53,36 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
         // Waits for patientPromises to be loaded
         const patientResults = await Promise.all(patientPromises)
 
-        const patientsData: Record<number, Patient | null> = {}
-
-        patientResults.forEach(({ id, patient }) => {
-          patientsData[id] = patient
+        setPatients((prevPatients) => {
+          const newPatientsData = { ...prevPatients }
+          patientResults.forEach(({ id, patient }) => {
+            newPatientsData[id] = patient
+          })
+          return newPatientsData
         })
```

Comment:
Functional programming for the win, how about:
```ts
setPatients(prevPatients => {
    return patientResults.reduce((newPatientsData, { id, patient }) => {
      return { ...newPatientsData, [id]: patient }
    }, { ...prevPatients })
})
```

Reply:
I've create a function